src/main/java/org/spongepowered/api/schematic/Schematic.java

```diff
@@ -0,0 +1,210 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.schematic;
+
+import com.flowpowered.math.vector.Vector3i;
+import com.google.common.collect.BiMap;
+import org.spongepowered.api.block.BlockSnapshot;
+import org.spongepowered.api.block.BlockState;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.DataSerializable;
+import org.spongepowered.api.data.persistence.DataBuilder;
+import org.spongepowered.api.data.persistence.InvalidDataException;
+import org.spongepowered.api.world.World;
+import org.spongepowered.api.world.extent.Extent;
+
+import java.util.Optional;
+
+/**
+ * Unless otherwise noted, "the specification" refers to the latest version of the Schematic Specification {@url https://github
+ * .com/SpongePowered/Schematic-Specification}.
+ */
+public interface Schematic extends DataSerializable {
+
+    /**
+     * Gets the attached {@link DataContainer}.
+     *
+     * <p>Correlates with the optional `Metadata` field of the specification.</p>
+     *
+     * @return The attached {@link DataContainer} or {@link Optional#empty()} if none
+     */
+    Optional<DataContainer> getMetadata();
+
+    /**
+     * Gets the width of the Schematic.
+     *
+     * <p>Correlates with the `Width` field of the specification.</p>
+     *
+     * @return The width of the Schematic.
+     */
+    int getWidth();
+
+    /**
+     * Gets the height of the Schematic.
+     *
+     * <p>Correlates with the `Height` field of the specification.</p>
+     *
+     * @return The height of the Schematic.
+     */
+    int getHeight();
+
+    /**
+     * Gets the length of the Schematic.
+     *
+     * <p>Correlates with the `Length` field of the specification.</p>
+     *
+     * @return The length of the Schematic.
+     */
+    int getLength();
+
+    /**
+     * Gets the origin of the Schematic (relative to the minimum point) or the minimum point if none.
+     *
+     * <p>Correlates with the `OffsetX`, `OffsetY`, and `OffsetZ` fields of the specification.</p>
+     *
+     * @return The origin of the Schematic
+     */
+    Vector3i getOrigin();
+
+    /**
+     * Gets the largest key in the palette.
+     *
+     * @return The largest key
+     */
+    int getPaletteMax();
+
+    /**
+     * Gets the palette of the Schematic.
+     *
+     * <p>Correlates backwards to the `Palette` field of the specification.</p>
+     *
+     * @return The palette
+     */
+    BiMap<Integer, BlockState> getPalette();
+
+    /**
+     * Gets the {@link BlockState} at the relative location, or {@link Optional#empty()} if none.
+     *
+     * @param extent The extent
+     * @param location The location (relative to the origin of the schematic) of the BlockSnapshot desired
+     * @return The BlockState found
+     */
+    Optional<BlockState> getBlockAt(Extent extent, Vector3i location);
```

Comment:
But, this method is just getting a `BlockState` stored in the schematic. What does the `extent` parameter even mean?